### ISSUE TITLE
[java] Rule to detect overly verbose array initializiation

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#1495]
E.g.
```java 
Foo[] x = new Foo[] { ... };
```
can be written equivalently as 
```java
Foo[] x = { ... }; 
 ```
Also, 
```java 
Foo[] x = new Foo[0];
// can be rewritten to 
Foo[] x = {}; // I think this should probably be opt-out, since new Foo[0] is very readable 
```

This is only possible in local variable or field declaration initializers, in all other contexts, the `new Foo[]` is mandatory. E.g. the following is disallowed:
```java
Foo[] x;
x = {...};
```

The rule would belong in the code style category.

I read on stackoverflow that this lighter syntax is not from the original Java 1.0 spec but probably from 1.1, but we don't have language versions below 1.3 anyway. Recent JLS: https://docs.oracle.com/javase/specs/jls/se8/html/jls-10.html#jls-10.6

**Comment by adangel** [comment]
Welcome @riteshmodi !
I've send you a github invite - once you accept, I'll assign the issue to you.

**Comment by IDoCodingStuffs** [comment]
@adangel I would like to tackle this in case @riteshmodi is not working on it

**Comment by jsotuyod** [comment]
@IDoCodingStuffs welcomed! We haven't heard back from @riteshmodi nor has he accepted the intive. So you are more than welcomed to take this over!

**Comment by IDoCodingStuffs** [comment]
Awesome, will start working on it then
